bootstrap/kubeadm/main.go

```diff
@@ -151,11 +152,11 @@ func main() {
 
 	// klog.Background will automatically use the right logger.
 	ctrl.SetLogger(klog.Background())
-
 	if profilerAddress != "" {
 		klog.Infof("Profiler listening for requests at %s", profilerAddress)
 		go func() {
-			klog.Info(http.ListenAndServe(profilerAddress, nil))
+			srv := http.Server{Addr: profilerAddress, ReadHeaderTimeout: 2 * time.Second}
+			klog.Info(srv.ListenAndServe())
```

Comment:
Can we change the log to do the same as for manager? e.g.
```go
			if err := srv.ListenAndServe(); err != nil {
				klog.ErrorS(err, "problem running profiler server")
			}
```

P.S. I don't know why klog is used here instead of setupLog, maybe there's a reason. (I'm fine with just keep using klog in this go routine)

Reply:
To use Setuplog we have to do 
```suggestion
	                setupLog.Info(srv.ListenAndServe().Error())
```

I think it's fine to leave as is though.